### ISSUE TITLE
New Rule: no-trailing-spaces

### ISSUE THREAD

**Issue by BHyeonKim** [eslint/markdown#735]
### Rule details

Disallow trailing whitespace at the end of lines in Markdown content, except where it is used intentionally to create a hard line break. 

In CommonMark, a line ending preceded by 2 or more spaces produces a hard break (`<br>`). Anywhere else, trailing whitespace has no effect on rendering.    

This rule allows exactly two trailing-space counts and flags everything else:  
- 0 trailing spaces: allowed (no effect, nothing to flag).   
- 1 trailing space: incorrect — has no effect.   
- 2 trailing spaces: allowed (intentional hard break).   
- 3 or more trailing spaces: incorrect — exceeds what's needed for a hard break.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- Correct -->
This line has no trailing whitespace.[0 space]

This line ends with a hard break.[2 spaces]
Next line.

<!-- Incorrect -->
This line has one trailing space.[1 space]

This line has three trailing spaces.[more than 2 spaces]
Next line.
```

### Prior Art

markdownlint - [md009](https://github.com/DavidAnson/markdownlint/blob/main/doc/md009.md)

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca): confirmed.

Thanks for the issue. Although some might think this rule is stylistic, I believe it is about hard-break syntax detection and preventing potential mistakes, so I’m in favor of it.

Since this is a feature, I’m leaving it open for others to review as well.

**Comment by DMartens** [comment]
I think the general case of trailing spaces would be stylistic (and as such should not be a core rule).
We could focus the rule only on trailing spaces inside paragraphs and report for any whitespace different from 2 spaces:
```md
single space 
three spaces   
tabs	
```

**Comment by BHyeonKim** [comment]
I agree with @DMartens. A rule that reports any trailing whitespace at the end of a line other than exactly 2 spaces sounds good.

**Comment by lumirlumir** [comment]
I’m not sure using a different approach from the other existing rules makes sense here.

There was a [discussion about the `checkClosedHeadings` option in the `no-missing-atx-heading-space` rule](https://github.com/eslint/markdown/issues/409#issuecomment-2959423927). Although that option is a stylistic choice, since closed ATX headings are a stylistic choice, the behavior was accepted to ensure a seamless transition from `markdownlint` and remain aligned with its existing behavior.

So, in this case, I think it would be better to align with the existing linter rule’s behavior.
